pkg/hubble/relay/server/server.go

```diff
@@ -99,39 +124,13 @@ func New(options ...Option) (*Server, error) {
 // Stop() is not called.
 func (s *Server) Serve() error {
 	s.opts.log.WithField("options", fmt.Sprintf("%+v", s.opts)).Info("Starting server...")
-
-	switch {
-	case s.opts.insecureServer:
-		s.server = grpc.NewServer()
-	case s.opts.serverTLSConfig != nil:
-		tlsConfig := s.opts.serverTLSConfig.ServerConfig(&tls.Config{
-			MinVersion: MinTLSVersion,
-		})
-		creds := credentials.NewTLS(tlsConfig)
-		s.server = grpc.NewServer(grpc.Creds(creds))
-	default:
-		return ErrNoServerTLSConfig
-	}
```

Comment:
This error message is removed by the refactor. Please add it back in to `New()`. I believe this error is thrown intentionally, such that forgetting the TLS config cannot be done without explicitly requesting an insecure server.

Reply:
Good catch 👍

Reply:
This should be handled here I believe? https://github.com/cilium/cilium/pull/20376/files#diff-c559920792798792688868000d2e056dc5ca3eed9c7a973ae73e8aea39e0bd7dL54-L59

Reply:
The link somehow didn't work for me, but judging from the URL you likely were referring to this:
https://github.com/cilium/cilium/blob/5b51ba8784ee30d9de85212fff8283ac93a4da81/pkg/hubble/relay/server/server.go#L56-L58

Yup, agreed that this should be handled. Thanks!

Reply:
Such is the pain of GitHub reviews and force pushing sometimes.